metricbeat/module/postgresql/statements/statements.go

```diff
@@ -0,0 +1,57 @@
+package statements
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/module/postgresql"
+
+	// Register postgresql database/sql driver
+	_ "github.com/lib/pq"
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	mb.Registry.MustAddMetricSet("postgresql", "statements", New,
+		mb.WithHostParser(postgresql.ParseURL),
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet type defines all fields of the Postgresql MetricSet
+type MetricSet struct {
+	mb.BaseMetricSet
+}
+
+// New create a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
```

Comment:
Add this metricset as beta, for that print a warning here with something like:
```
cfgwarn.Beta("The statement metricset is beta")
```
And change release to beta in fields.yml.